docs/doc-system.md

```diff
@@ -0,0 +1,114 @@
+Slang Doc System 
+================
+
+Slang contains a rudimentary documentation generation system. The mechanism used to mark up source is similar to [doxygen](https://www.doxygen.nl/manual/docblocks.html). Namely
+
+```
+/**
+ ... text ... (JavaDoc style)
+ */
+void someFunctionA() {} 
+ 
+/*!
+ .. text .. (QT style)
+ another line
+ */ 
+void someFunctionB() {} 
+ 
+/// ... text ... (Multi line)
+/// another line
+void someFunctionC() {}
+ 
+//!... text ...  (QT Multi line)
+//! another line
+void someFunctionD() {}
+ 
+```
+ 
+All of the above examples will add the documentation for the declaration that appears after them. Also note that this slightly diverges from doxygen in that an empty line before and after in a multi line comment is *not* required.
+
+We can also document the parameters to a function similarly
+
+```
+/// My function
+void myFunction(
+    /// The A parameter 
+    int a, 
+    /// The B parameter
+    int b);
+```
+
+If you just need a single line comment to describe something, you can place the documentation after the parameter as in
+
+```
+
+/// My function
+void myFunction(    int a,      //< The A parameter     
+                    int b)      //< The B parameter
```

Comment:
It isn't related to the documentation, but I'd like for us to eventually allow this kind of syntax with ordinary `///` comments and not needing `//<`. It is "obvious" to a reader of the code that a trailing comment like this pertains to the preceding declaration and not the next one.

If we just using the leading/trailing trivia definition, then we can say that a comment like these is trailing trivia on the declaration (or the `,` / `)` after it), and thus belongs to the declaration.

(Yes, I have an agenda here to get rid of all the `<`, `!`, etc. nonsense that doxygen pushes onto people. I want our solution to first and foremost be something we can describe as "humane")

Reply:
That without some extra heuristics/limitations this introduces an ambiguity.  Take

```
enum Blah
{
   A, /// Some documentation
   B, 
};
```
Is it B or As? You could say ah - because it's after A *on the same line* it belongs to A. Ok now

```
enum Blah
{ 
   A,  /// A 
        /// multi-line comment 
   
   B,
};
```

That //< allows (fairly simply) multi line is the point here. 
I don't bring this up to make out this is some insurmountable challenge, just that given the features of the doxygen style/s it isn't without reasonable reason that they have the //< and associated styles. 

I admit I perhaps also do find the //< style particularly objectionable having used it in many code bases for many years.